git-hub

```diff
@@ -796,6 +797,37 @@ class CloneCmd (object):
 		git_quiet(1, 'fetch', remote)
 
 	@classmethod
+	def validate_url(cls, url, urltype):
+		scheme = urltype[:-4] # Remove the '_url'
+		# Any URL starting with <transport>:: will run
+		# a git-remote-<transport> command. "ext::" is specially
+		# unsafe, as that can run arbitrary commands.
+		if '::' in url:
+			pass # Fishy already, so skip to the question
+		elif scheme in ('clone', 'svn') and re.match(r'^https?://', url):
+			return
+		# XXX: git clone man page says it should be "git://",
+		# but github uses "git:"... I don't know how one is supposed to
+		# distinguish between an SSH URL without the "ssh://" and "git"
+		# as the host and this scheme... but well, accepting this as
+		# long as git knows how to deal with it should be safe.
+		elif scheme == 'git' and re.match(r'^git:', url):
+			return
```

Comment:
Mmmm, then there might be just outdated (or wrong) [documentation examples](https://developer.github.com/v3/repos/#get):
![git](https://cloud.githubusercontent.com/assets/2684688/19123856/cfeb627a-8b30-11e6-83c3-24f8f00b95bd.png)